tiledb/sm/filesystem/azure.h

```diff
@@ -283,33 +283,50 @@ class Azure {
 
   class AzureRetryPolicy final : public azure::storage_lite::retry_policy_base {
    public:
-    AzureRetryPolicy(const bool under_test)
-        : azure::storage_lite::retry_policy_base()
-        , under_test_(under_test) {
-    }
-
+    /**
+     * The SDK invokes this routine before each request to Azure. This returns
+     * a pair: a bool to indicate if we should make a request and a time
+     * interval to wait before starting the request.
+     */
     azure::storage_lite::retry_info evaluate(
         const azure::storage_lite::retry_context& context) const override {
-      const int32_t max_retries = under_test_ ? 1 : 3;
+      const int http_code = context.result();
 
-      // The SDK consults this return value for the initial request. We must
-      // return an instance of 'retry_info' with a 0-delay wait time to prevent
-      // the SDK from sleeping on the first request.
-      if (context.numbers() < 1) {
+      // When the response code is 0, the SDK has yet to send the initial
+      // request. Allow the request to start immediately by returning a 0-second
+      // delay.
+      if (http_code == 0) {
         return azure::storage_lite::retry_info(true, std::chrono::seconds(0));
       }
 
+      // Determine if we should retry on the returned http code in the response.
+      if (!should_retry(http_code)) {
+        return azure::storage_lite::retry_info(false, std::chrono::seconds(0));
+      }
+
       // Wait one second before all retry attempts.
+      static const int32_t max_retries = 3;
       if (context.numbers() < max_retries) {
         return azure::storage_lite::retry_info(true, std::chrono::seconds(1));
```

Comment:
I'm wondering why 1s here? Maybe the conditions/paths/etc are different, but IIUC we configure S3 to sleep for 100 ms:

```
tiledb/sm/misc/constants.cc
455:const unsigned int s3_attempt_sleep_ms = 100;
```

Reply:
Totally arbitrary, I'll modify this to do 100ms too.

Reply:
The SDK only takes retry intervals in the 1-second granularity. I've added constants but adjusted it for a 1-second retry interval. If we're really concerned about this we can patch the SDK.

Reply:
Ok, nevermind for now. Thanks for checking.